lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -112,6 +112,17 @@ class AwsCompileFunctions {
         throw new Error(`No artifact path is set for function: ${functionName}`);
       }
 
+      if (this.serverless.service.provider.deploymentBucket) {
+        const deploymentBucket = this.serverless.service.provider.deploymentBucket;
```

Comment:
Validation of the bucket name has been moved to configureStack.  That puts the CloudFormation setup in one place.